### PR TITLE
Add WMT20 MLQE 3 shared tasks

### PR DESCRIPTION
**Pull request by VictorSanh**
3 tasks for the WMT 20 MLQE shared tasks -> 3 different datasets

(I re-created #1137 because it was too messy).

Note that in L199 `task3.py`, I used `logging.warning` to print some missing data in the train set.

### CODE REVIEW

File: datasets/wmt_mlqe_task2/README.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+*Task 1 evaluates the application of QE for post-editing purposes. It consists of predicting:*
+- ***Word-level tags.** This is done both on source side (to detect which words caused errors) and target side (to detect mistranslated or missing words).*
+  - ***Target.** Each token is tagged as either `OK` or `BAD`. Additionally, each gap between two words is tagged as `BAD` if one or more missing words should have been there, and `OK` otherwise. Note that number of tags for each target sentence is 2*N+1, where N is the number of tokens in the sentence.*
+  - ***Source.** Tokens are tagged as `OK` if they were correctly translated, and `BAD` otherwise. Gaps are not tagged.*
+- ***Sentence-level HTER scores.** HTER (Human Translation Error Rate) is the ratio between the number of edits (insertions/deletions/replacements) needed and the reference translation length.*
```
Comment (lhoestq) [inline datasets/wmt_mlqe_task2/README.md:65]: the markdown doesn't seem to be parsed correctly by github at this point, can you check that your're not missing `*` somewhere ?

File: datasets/wmt_mlqe_task2/wmt_mlqe_task2.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    ]
+
+    DEFAULT_CONFIG_NAME = "en-de"
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "src": datasets.Value("string"),
+                "src_tags": datasets.Sequence(datasets.ClassLabel(names=["BAD", "OK"])),
+                "mt": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/wmt_mlqe_task2/wmt_mlqe_task2.py:105]: can you use the `Translation` feature type here as well ?

File: datasets/wmt_mlqe_task1/wmt_mlqe_task1.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+            version=datasets.Version("1.1.0"),
+            description=f"Task 1: {src_lg} - {tgt_lg}",
+            src_lg=src_lg,
+            tgt_lg=tgt_lg,
+        )
+        for (src_lg, tgt_lg) in _LANGUAGE_PAIRS
+    ]
+
+    DEFAULT_CONFIG_NAME = "en-de"
+
```
Comment (lhoestq) [inline datasets/wmt_mlqe_task1/wmt_mlqe_task1.py:100]:
```suggestion
    BUILDER_CONFIG_CLASS = WmtMlqeConfig

```
we need that to specify that we're using a custom config

File: datasets/wmt_mlqe_task1/README.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+- found
+languages:
+- de
+- en
+- et
+- ne
+- ro
+- ru
+- si
+- zh
```
Comment (lhoestq) [inline datasets/wmt_mlqe_task1/README.md:15]: The languages here depend on the configuration, so it must be formatted by providing the set of languages per config. For example
```
languages:
  en-de:
  - en
  - de
  en-ru:
  - en
  - ru
```
(must be filled for each config)

Can you change that please ?

same for task2

File: datasets/wmt_mlqe_task3/wmt_mlqe_task3.py [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+
+        def open_and_read(fp):
+            with open(fp, encoding="utf-8") as f:
+                return f.read().splitlines()
+
+        for id_, folder in enumerate(glob.glob(f"{filepath}/*")):
```
Comment (lhoestq) [inline datasets/wmt_mlqe_task3/wmt_mlqe_task3.py:192]:
```suggestion
        for id_, folder in enumerate(sorted(glob.glob(os.path.join(filepath, "*")))):
```
- make the order deterministic
- use os.path.join for windows users

File: datasets/wmt_mlqe_task3/wmt_mlqe_task3.py [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+            source_segments = open_and_read(os.path.join(folder, "source.segments"))
+            source_tokenized = open_and_read(os.path.join(folder, "source.tokenized"))
+            mt_segments = open_and_read(os.path.join(folder, "mt.segments"))
+            mt_tokenized = open_and_read(os.path.join(folder, "mt.tokenized"))
+
+            if split in ["train", "dev"] and not os.path.exists(os.path.join(folder, "token_index")):
+                logging.warning(
```
Comment (lhoestq) [inline datasets/wmt_mlqe_task3/wmt_mlqe_task3.py:199]: maybe just mention in the dataset card that some token_index are missing ?
we don't want to pollute the users' logs

File: datasets/wmt_mlqe_task3/wmt_mlqe_task3.py [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+        datasets.BuilderConfig(
+            name=f"plain_text",
+            version=datasets.Version("1.1.0"),
+            description=f"Plain text",
+        )
+    ]
+
+    def _info(self):
+        features = datasets.Features(
+            {
```
Comment (lhoestq) [inline datasets/wmt_mlqe_task3/wmt_mlqe_task3.py:118]: Here one example corresponds to one document
But it looks like each column correspond to the features for each segment.
Should we consider having one example = one segment instead ?

Let me know what you think

**Review by lhoestq — COMMENTED** [review]
Awesome thank you !

I left a few comments

File: datasets/wmt_mlqe_task3/wmt_mlqe_task3.py (hunk 7 above)
Comment (VictorSanh) [reply datasets/wmt_mlqe_task3/wmt_mlqe_task3.py]: I think it is crucial to keep the document structure to be able to model the long-range dependencies.

**Comment by VictorSanh** [comment]
Thanks for the comments Quentin!
I integrated them

File: datasets/wmt20_mlqe_task1/wmt20_mlqe_task1.py [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+
+
+class WmtMlqeConfig(datasets.BuilderConfig):
+    def __init__(self, src_lg, tgt_lg, **kwargs):
+        super(WmtMlqeConfig, self).__init__(**kwargs)
+        self.src_lg = src_lg
+        self.tgt_lg = tgt_lg
+
+
+class WmtMlqeTask1(datasets.GeneratorBasedBuilder):
```
Comment (lhoestq) [inline datasets/wmt20_mlqe_task1/wmt20_mlqe_task1.py:85]:
```suggestion
class Wmt20MlqeTask1(datasets.GeneratorBasedBuilder):
```
This must be the camel case version of the dataset script name `wmt20_mlqe_task1`

File: datasets/wmt20_mlqe_task1/dataset_infos.json [hunk 9]
```diff
@@ -0,0 +1 @@
...nmt_output": {"dtype": "string", "id": null, "_type": "Value"}, "word_probas": {"feature": {"dtype": "float32", "id": null, "_type": "Value"}, "length": -1, "id": null, "_type": "Sequence"}}, "post_processed": null, "supervised_keys": null, "builder_name": "wmt_mlqe_task1", "config_name": "ru-en", "version": {"version_str": "1.1.0", "description": null, "major": 1, "minor": 1, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 3520776, "num_examples": 7000, "dataset_name": "wmt_mlqe_task1"}, "test": {"name": "test", "num_bytes": 478787, "num_examples": 1000, "dataset_name": "wmt_mlqe_task1"}, "validation": {"name": "validation", "num_bytes": 499601, "num_examples": 1000, "dataset_name": "wmt_mlqe_task1"}}, "download_checksums": {"https://www.quest.dcs.shef.ac.uk/wmt20_files_qe/ru-en.tar.gz": {"num_bytes": 1897111, "checksum": "a1b762e4c21195b3c8e65e65ab5fb75527d4f8d3ce54e34de18ef773d22f675f"}, "https://www.quest.dcs.shef.ac.uk/wmt20_files_qe/ru-en_test.tar.gz": {"num_bytes": 226573, "checksum": "9c1d35a2b3f5d807f5fa54ae9114ba8384dcd2929452435b54306efe1d501956"}}, "download_size": 2123684, "post_processing_size": null, "dataset_size": 4499164, "size_in_bytes": 6622848}}
```
Comment (lhoestq) [inline datasets/wmt20_mlqe_task1/dataset_infos.json:1]: Can you regenerate this file after the class name change please ?

Also it looks like it has metadata about previous config names that you may have generated in the past.
So please delete this file and clear your cache before regenerating it.
For example the config `en-de-task1` doesn't exist anymore

File: datasets/wmt20_mlqe_task2/wmt20_mlqe_task2.py [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+
+
+class WmtMlqeConfig(datasets.BuilderConfig):
+    def __init__(self, src_lg, tgt_lg, **kwargs):
+        super(WmtMlqeConfig, self).__init__(**kwargs)
+        self.src_lg = src_lg
+        self.tgt_lg = tgt_lg
+
+
+class WmtMlqeTask2(datasets.GeneratorBasedBuilder):
```
Comment (lhoestq) [inline datasets/wmt20_mlqe_task2/wmt20_mlqe_task2.py:84]:
```suggestion
class Wmt20MlqeTask2(datasets.GeneratorBasedBuilder):
```

File: datasets/wmt20_mlqe_task3/wmt20_mlqe_task3.py [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+    "Wrong Auxiliary Verb",
+    "Wrong Conjunction",
+    "Wrong Determiner",
+    "Wrong Language Variety",
+    "Wrong Preposition",
+    "Wrong Pronoun",
+]
+
+
+class WmtMlqeTask1(datasets.GeneratorBasedBuilder):
```
Comment (lhoestq) [inline datasets/wmt20_mlqe_task3/wmt20_mlqe_task3.py:104]:
```suggestion
class Wmt20MlqeTask3(datasets.GeneratorBasedBuilder):
```

**Review by lhoestq — COMMENTED** [review]
Really cool thanks :) 

I just noticed that the dataset_infos.json file of the task1 needs to be regenerated

**Comment by VictorSanh** [comment]
It should be ok now!
Sorry I wasn't attentive enough.
(tests are currently failing, I understand it's from other datasets)

**Review by lhoestq — APPROVED** [review]
Perfect thank you !

**Comment by lhoestq** [comment]
merging since the CI is fixed on master

**Pull request merged**